apis/infix-en_GB/extensions/jdk8/atrium-api-infix-en_GB-jdk8-jvm/src/main/kotlin/ch/tutteli/atrium/api/infix/en_GB/jdk8/pathAssertions.kt

```diff
@@ -0,0 +1,263 @@
+@file:Suppress("JAVA_MODULE_DOES_NOT_READ_UNNAMED_MODULE" /* TODO remove once https://youtrack.jetbrains.com/issue/KT-35343 is fixed */)
+
+package ch.tutteli.atrium.api.infix.en_GB.jdk8
+
+import ch.tutteli.atrium.creating.Expect
+import ch.tutteli.atrium.domain.builders.ExpectImpl
+import ch.tutteli.atrium.domain.builders.path
+import java.nio.file.Path
+
+/**
+ * Expects that the subject of the assertion (a [Path]) starts with the [expected] [Path].
+ *
+ * @return This assertion container to support a fluent API.
+ * @throws AssertionError Might throw an [AssertionError] if the assertion made is not correct.
+ *
+ * @since 1.0.0
+ */
+infix fun <T : Path> Expect<T>.startsWith(expected: Path): Expect<T> =
+    addAssertion(ExpectImpl.path.startsWith(this, expected))
+
+/**
+ * Expects that the subject of the assertion (a [Path]) does not start with the [expected] [Path].
+ *
+ * @return This assertion container to support a fluent API.
+ * @throws AssertionError Might throw an [AssertionError] if the assertion made is not correct.
+ *
+ * @since 1.0.0
+ */
+infix fun <T : Path> Expect<T>.startsNotWith(expected: Path): Expect<T> =
+    addAssertion(ExpectImpl.path.startsNotWith(this, expected))
+
+/**
+ * Expects that the subject of the assertion (a [Path]) ends with the expected [Path].
+ *
+ * @return This assertion container to support a fluent API.
+ * @throws AssertionError Might throw an [AssertionError] if the assertion made is not correct.
+ *
+ * @since 1.0.0
+ */
+infix fun <T : Path> Expect<T>.endsWith(expected: Path): Expect<T> =
+    addAssertion(ExpectImpl.path.endsWith(this, expected))
+
+/**
+ * Expects that the subject of the assertion (a [Path]) does not end with the expected [Path];
+ *
+ * @param expected The [Path] provided to the assertion
+ * @return This assertion container to support a fluent API.
+ * @throws AssertionError Might throw an [AssertionError] if the assertion made is not correct.
+ *
+ * @since 1.0.0
+ */
+infix fun <T : Path> Expect<T>.endsNotWith(expected: Path): Expect<T> =
+    addAssertion(ExpectImpl.path.endsNotWith(this, expected))
+
+/**
+ * Expects that the subject of the assertion (a [Path]) exists;
+ * meaning that there is a file system entry at the location the [Path] points to.
+ *
+ * This matcher _resolves_ symbolic links. Therefore, if a symbolic link exists at the location the subject points to,
+ * then the search will continue at that location.
+ *
+ * @return This assertion container to support a fluent API.
+ * @throws AssertionError Might throw an [AssertionError] if the assertion made is not correct.
+ *
+ * @since 1.0.0
+ */
+infix fun <T : Path> Expect<T>.does(@Suppress("UNUSED_PARAMETER") exist: exist): Expect<T> =
```

Comment:
maybe better `toBe existing` to fit into the schema of the other functions?

Reply:
I thought about that as well. But I felt that `expect(x) toBe existing` is just very awkward phrasing. `expect(x) does exist` feels more natural to me.

But I can change it if you prefer `toBe existing`. 

From a technical perspective, you can make an argument for both:

 * It’s easier to discover `toBe existing` if I already know `toBe writable`
 * However, if a lot of assertions share the same prefix, code completion becomes a lot less useful, because I can quickly select the function by typing.

Reply:
how about `to exist` instead? We don't have `does` as prefix yet. But up to you

Reply:
Ah, of course! `to exist` is good.